extension/src/saturation.rs

```diff
@@ -0,0 +1,114 @@
+use pgx::*;
+
+/// Saturating Math for Integers
+
+/// Computes x+y, saturating at the numeric bounds instead of overflowing
+#[pg_extern(schema = "toolkit_experimental", immutable, parallel_safe)]
+fn addition_saturate_to_min(x: i32, y: i32) -> i32 {
```

Comment:
I'd go with `saturating_add` here.  It's more succinct and makes it more obvious that it's saturating on both high and low ends.